lib/active_merchant/billing/gateways/visanet_peru.rb

```diff
@@ -0,0 +1,239 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class VisanetPeruGateway < Gateway
+      self.display_name = "VisaNet Peru Gateway"
+      self.homepage_url = "http://www.visanet.com.pe"
+
+      self.test_url = "https://devapi.vnforapps.com/api.tokenization/api/v2/merchant"
+      self.live_url = "https://api.vnforapps.com/api.tokenization/api/v2/merchant"
+
+      self.supported_countries = ["US", "PE"]
+      self.default_currency = "PEN"
+      self.money_format = :cents
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover]
+
+      def initialize(options={})
+        requires!(options, :login, :password)
+        super
+      end
+
+      def purchase(amount, payment_method, options={})
+        MultiResponse.run() do |r|
+          r.process { authorize(amount, payment_method, options) }
+          r.process { capture(r.authorization, options) }
+        end
+      end
+
+      def authorize(amount, payment_method, options={})
+        #JSON
+        post = {}
+
+        add_invoice(post, amount, options)
+        add_payment_method(post, payment_method)
+        add_customer_data(post, options)
+        add_device_fingerprint_data(post, options)
+        add_merchant_define_data(post, options)
+
+        # No vaulting for now
+        post[:createAlias] = false
+
+        commit("authorize", post, options)
+      end
+
+      def capture(authorization, options={})
+        params = {}
+        action, merchant_id, purchase_number = split_authorization(authorization)
+        options[:merchant_id] = merchant_id
```

Comment:
Usually we treat `options` as a read-only hash. So instead of assigning `merchant_id` to a new option, just add it to the `params` hash.